InnerEye/ML/utils/io_util.py

```diff
@@ -412,24 +412,43 @@ def load_image_in_known_formats(file: Path,
         raise ValueError(f"Unsupported image file type for path {file}")
 
 
-def load_labels_from_dataset_source(dataset_source: PatientDatasetSource, check_exclusive: bool = True) -> np.ndarray:
+def load_labels_from_dataset_source(dataset_source: PatientDatasetSource, check_exclusive: bool = True,
+                                    mask_size: Optional[Tuple[int]] = None) -> np.ndarray:
     """
     Load labels containing segmentation binary labels in one-hot-encoding.
     In the future, this function will be used to load global class and non-imaging information as well.
 
+    :type mask_size: Image size, tuple if integers.
     :param dataset_source: The dataset source for which channels are to be loaded into memory.
     :param check_exclusive: Check that the labels are mutually exclusive (defaults to True)
     :return: A label sample object containing ground-truth information.
     """
-    labels = np.stack(
-        [load_image(gt, ImageDataType.SEGMENTATION.value).image for gt in dataset_source.ground_truth_channels])
 
-    if check_exclusive and (sum(labels) > 1.).any():  # type: ignore
+    if not dataset_source.allow_incomplete_labels:
+        labels = np.stack(
+            [load_image(gt, ImageDataType.SEGMENTATION.value).image for gt in dataset_source.ground_truth_channels])
+    else:
+        assert mask_size is not None
+        label_list = []
+        for gt in dataset_source.ground_truth_channels:
+            if str(gt) == '.':
```

Comment:
This is why I'm not a fan of using Path("") as a magical value, see the above comment. Your code now effectively relies on the internals of how Path.__str__ works.